### PR TITLE
Page & action to (re-)invoke upgrade code methods

### PR DESCRIPTION
**Pull request by labkey-adam**
#### Rationale
[Issue 43665: Provide action to invoke arbitrary upgrade code methods](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=43665)

image

### CODE REVIEW

**Review by labkey-klum — APPROVED** [review]

File: core/src/org/labkey/core/admin/sql/SqlScriptController.java [hunk 1]
```diff
@@ -1300,6 +1308,111 @@ public void addNavTrail(NavTree root)
...
+            return _module;
+        }
+
+        public String getMethod()
+        {
+            return _method;
+        }
+    }
+
+    @RequiresPermission(AdminOperationsPermission.class)
```
Comment (labkey-jeckels) [inline core/src/org/labkey/core/admin/sql/SqlScriptController.java:1344]: Should we make this restricted to Site Admin instead? I can't imagine wanting to have hosted customers invoke this on their own.

**Review by labkey-jeckels — APPROVED** [review]

**Review by labkey-susanh — APPROVED** [review]

File: core/src/org/labkey/core/admin/sql/SqlScriptController.java (hunk 1 above)
Comment (labkey-adam) [reply core/src/org/labkey/core/admin/sql/SqlScriptController.java:1344]: Sure... I had it site admin at first, but noticed that Cory had annotated all the other script actions with AdminOperations. (Though most of the links aren't shown in production mode.)

Comment (labkey-adam) [reply core/src/org/labkey/core/admin/sql/SqlScriptController.java:1344]: Wait... isn't `AdminOperationsPermission` available to site administrators only?

Comment (labkey-jeckels) [reply core/src/org/labkey/core/admin/sql/SqlScriptController.java:1344]: Hmm... You're right. I guess don't understand the difference between SiteAdminPermission and AdminOperationsPermission, but it looks like both are granted only to Site Admins

Comment (labkey-adam) [reply core/src/org/labkey/core/admin/sql/SqlScriptController.java:1344]: Yeah, me neither

**Pull request merged**